### ISSUE TITLE
import statements show not found

### ISSUE THREAD

**Issue by zzwx** [devkanro/intellij-protobuf-plugin#4]
Environment:
```
GoLand 2021.1
Build #GO-211.6693.119, built on April 6, 2021
Runtime version: 11.0.10+9-b1341.35 amd64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1966M
Cores: 32
Registry: ide.completion.variant.limit=500, documentation.show.toolbar=true, debugger.watches.in.variables=false, suggest.all.run.configurations.from.context=true, ide.balloon.shadow.size=0, go.run.processes.with.pty=false, eslint.additional.file.extensions=svelte
Non-Bundled Plugins: Rider UI Theme Pack (0.8), coderpillr.theme (1.0.0-rc.3), com.alicannklc.tailwind (1.4.6), com.bum.intellij.tailwindcss (1.1.4), com.github.ushiosan23.darktheme (1.0.2), com.intellij.ideolog (203.0.27.0), com.intellij.plugins.vscodekeymap (211.4961.30), com.jetbrains.darkPurpleTheme (1.2), org.jetbrains.plugins.less (211.6693.76), com.mfwebstudio.atomonedark.plugin.id (1.0.3), com.vecheslav.darculaDarkerTheme (1.2.0), com.xinkun.theme.md (1.7.12), github.cweijan.test-theme (1.6), io.kanro.idea.plugin.protobuf (0.0.3), net.antelle.intellij-xcode-dark-theme (1.2.2), PythonCore (211.6693.119)
```

As an example, create 2 files:

`util.proto`:
```proto
syntax = "proto3";
package test;

option go_package = "test/grpc/pb";

message Sort {
    string field = 1;
    int64 order = 2; // 1 standing for asc, -1 for desc
}

message OffsetLimit {
    int64 offset = 1;
    int64 limit = 2;
}
```

and another one that uses it:

`test.proto`:
```proto
syntax = "proto3";
package test;

import "google/protobuf/timestamp.proto";
import "util.proto";

option go_package = "test/grpc/pb";

service Updater {
    rpc Update (UpdateRequest) returns (UpdateRequestReply);
}

message UpdateRequest {
    repeated Sort order_field = 1;
}

message UpdateRequestReply {
    string error                           = 1;
    google.protobuf.Timestamp last_updated = 2;
}
```

Monitor that the editor has no idea what Sort is coming from.

Regarding `google.protobuf.Timestamp`, as it is a global to proto, I have no idea how the plugin is supposed to find location of definitions.

**Comment by devkanro** [comment]
I'm developing the plugin base on the java project, plugin will scan `.proto` file in all source root and library root.
I will add some setting pages to configure the external import path. 
I'll have a check about how the plugin resolving import in GoLand, I will comment here if I got any updates.

**Comment by zzwx** [comment]
Thank you. It seems not seeing even the current folder imports for some
reason.

**Comment by devkanro** [comment]
I create a setting page for configuring import roots of protobuf in commit 8bb8ed809d534c35ca161e66828c4379142fdb87.
You can add your import path in idea setting page `Language & Framework > Protobuf`
It will release soon in the next version 0.0.4
![](https://user-images.githubusercontent.com/9367842/114317352-43a18780-9b3a-11eb-8888-3440f3563fad.png)

**Comment by devkanro** [comment]
I want to auto-detect protos in library just same as java project.
But it seems go modules not contains the raw proto file.
All raw protos will be as resource included in jar file in java project, so the plugin can import them without any operation.

In your case, I think you need to download all library protos(etc `google/protobuf/timestamp.proto`) to a folder and add them to the setting.

**Comment by zzwx** [comment]
Thank you for your fast response!

Right, Java is simplifying the process most likely.

Regarding Go setup, if you could make the same-folder imports working, that would be already a great improvement! This way all google imports could be simply copied into the folder where all the rest of *.proto files reside. At least that would be an option for some.

A project-wise setting seems to be a valid workaround as well, especially if the folders could be specified as **relative** to the project root as an option.

**Comment by devkanro** [comment]
The same folder is not a good choice for it, I am trying to make a behave of plugin same as `protoc` compiler.

The plugin will auto-resolve the project root in 0.0.4, you can have a try when it released.

**Comment by zzwx** [comment]
Well I meant `import "util.proto"` as importing it from the same folder according to spec https://developers.google.com/protocol-buffers/docs/proto3#importing_definitions

**Comment by devkanro** [comment]
Anyway, resolve files in project root is a reasonable approach. And using the project settings in other case as fallback.

**Comment by zzwx** [comment]
Testing `v0.0.4`, it's looking good!
Editing individual path doesn't trigger refresh, but this is a next step into this great plugin. Well done!

I'm hoping one day things like `option go_package = "...";` will also be recognized as valid. No rush though!

Thank you again!  I've updated my review at https://plugins.jetbrains.com/plugin/16422-protobuf

**Comment by devkanro** [comment]
Thanks,
I am happy to hear the plugin can help your work, but there are still some bugs in 0.0.4 (import optimizer not working).
I have already released 0.0.5. for hotfix.
